### PR TITLE
Remove Provider Deprecations in Sqlite

### PR DESCRIPTION
**Pull request by jscheffl**
In Airflow 3 Dev Call we discussed and made a LAZY CONSENSUS to remove all deprecation's in providers prior 2.11 release in https://lists.apache.org/thread/lhy7zhz8yxo3jjpln0ds8ogszgb9b469.

This PR is for the provider Sqlite

Relates to https://github.com/apache/airflow/issues/44559

### CODE REVIEW

**Comment by gopidesupavan** [comment]
Can you please also update the docs? https://github.com/apache/airflow/blob/main/docs/apache-airflow-providers-sqlite/operators.rst?plain=1#L20

**Review by gopidesupavan — APPROVED** [review]

File: docs/apache-airflow-providers-sqlite/operators.rst [hunk 1]
```diff
@@ -1,69 +0,0 @@
...
-    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
-    KIND, either express or implied.  See the License for the
-    specific language governing permissions and limitations
-    under the License.
-
-
-
-.. _howto/operator:SqliteOperator:
-
-SQLExecuteQueryOperator to connect to Sqlite
```
Comment (eladkal) [inline docs/apache-airflow-providers-sqlite/operators.rst]: Why are we removing this?
It explains that to query SqlLite you should use SQLExecuteQueryOperator. That is information we should keep in the docs. Actually not only here but for all sql related providers that we replaced with SQLExecuteQueryOperator

Comment (gopidesupavan) [reply docs/apache-airflow-providers-sqlite/operators.rst]: Do we still need to keep deprecated doc information `_howto/operator:SqliteOperator` ? is change log is enough?

Comment (eladkal) [reply docs/apache-airflow-providers-sqlite/operators.rst]: File name and titles may be adjusted but I think its nice to have example and information how to use SQLExecuteQueryOperator with SqlLite (and same for all others.. trino, presto, etc...)

It's also OK to cross refrence to SQLExecuteQueryOperator docs.

Comment (jscheffl) [reply docs/apache-airflow-providers-sqlite/operators.rst]: Yeah, you are right - I deleted too much docs. Reverted and just changed the note.

File: providers/tests/sqlite/operators/test_sqlite.py [hunk 2]
```diff
@@ -1,85 +0,0 @@
...
-from airflow.utils import timezone
-
-DEFAULT_DATE = timezone.datetime(2015, 1, 1)
-DEFAULT_DATE_ISO = DEFAULT_DATE.isoformat()
-DEFAULT_DATE_DS = DEFAULT_DATE_ISO[:10]
-TEST_DAG_ID = "unit_test_dag"
-
-
-@pytest.mark.backend("sqlite")
-class TestSqliteOperator:
```
Comment (eladkal) [inline providers/tests/sqlite/operators/test_sqlite.py:33]: I wonder if we need to keep these tests?
How do we know SQLExecuteQueryOperator works with sqlite backend if we remove these?
Do we have this coverage elsewhere?

Comment (jscheffl) [reply providers/tests/sqlite/operators/test_sqlite.py:33]: We have pytests for `SQLExecuteQueryOperator` and these tests are executed against all DBs we support. I did not check in detail but if you think DB tests are missing... we should add it there.

Comment (eladkal) [reply providers/tests/sqlite/operators/test_sqlite.py:33]: I will check this

**Pull request merged**